server/channels/app/integration_action.go

```diff
@@ -532,3 +532,48 @@ func (a *App) SubmitInteractiveDialog(c request.CTX, request model.SubmitDialogR
 
 	return &response, nil
 }
+
+func (a *App) LookupInteractiveDialog(c request.CTX, request model.SubmitDialogRequest) (*model.LookupDialogResponse, *model.AppError) {
+	url := request.URL
+	request.URL = ""
+	request.Type = "dialog_lookup"
+
+	b, err := json.Marshal(request)
+	if err != nil {
+		return nil, model.NewAppError("LookupInteractiveDialog", "app.lookup_interactive_dialog.json_error", nil, "", http.StatusBadRequest).Wrap(err)
+	}
+
+	// Log request, regardless of whether destination is internal or external
+	c.Logger().Info("LookupInteractiveDialog POST request, through DoActionRequest",
+		mlog.String("url", url),
+		mlog.String("user_id", request.UserId),
+		mlog.String("channel_id", request.ChannelId),
+		mlog.String("team_id", request.TeamId),
+	)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(*a.Config().ServiceSettings.OutgoingIntegrationRequestsTimeout)*time.Second)
+	defer cancel()
+	resp, appErr := a.DoActionRequest(c.WithContext(ctx), url, b)
+	if appErr != nil {
+		return nil, appErr
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, model.NewAppError("LookupInteractiveDialog", "app.lookup_interactive_dialog.read_body_error", nil, "", http.StatusInternalServerError).Wrap(err)
+	}
```

Comment:
I have concerns with this. I understand it is following the existing pattern, but please see this thread:
https://hub.mattermost.com/private-core/pl/rwzp7dk3opnk5pk54uq9a3rijo